arch/xtensa/src/esp32/esp32_spi.c

```diff
@@ -861,11 +866,37 @@ static void esp32_spi_dma_exchange(FAR struct esp32_spi_priv_s *priv,
                                    uint32_t nwords)
 {
   uint32_t bytes = nwords * (priv->nbits / 8);
-  uint8_t *tp = (uint8_t *)txbuffer;
-  uint8_t *rp = (uint8_t *)rxbuffer;
+  uint8_t *tp;
+  uint8_t *rp;
   uint32_t n;
   uint32_t regval;
 
+  /* If the buffer comes from PSRAM, allocate a new one from DRAM */
+
+#ifdef CONFIG_XTENSA_USE_SEPERATE_IMEM
+  if (esp32_ptr_extram(txbuffer))
+    {
+      tp = up_imm_malloc(bytes);
+      DEBUGASSERT(tp != NULL);
```

Comment:
don't you need memcpy here?
(i don't understand what this function is doing. just a wild guess.)